backend/typescript/models/camp.model.ts

```diff
@@ -1,43 +1,55 @@
 import { Schema, model } from "mongoose";
-import { AbstractCamp } from "./abstractCamp.model";
+import { BaseCamp } from "./baseCamp.model";
+import { Camper } from "./camper.model";
 
-export interface Camp extends AbstractCamp {
+export interface Camp extends BaseCamp {
   baseCamp: Schema.Types.ObjectId;
-  campers: Schema.Types.ObjectId[];
+  campers: (Camper | Schema.Types.ObjectId)[];
+  capacity: number;
   waitlist: Schema.Types.ObjectId[];
-  startDate: Date;
-  endDate: Date;
+  dates: Date[];
+  startTime: string;
+  endTime: string;
   active: boolean;
 }
 
 const CampSchema: Schema = new Schema({
   baseCamp: {
     type: Schema.Types.ObjectId,
-    ref: "Camp",
+    ref: "BaseCamp",
     required: true,
   },
   campers: [
```

Comment:
ah I didn't catch this before, can we change the structure of both campers and waitlist? think this should be 
```
campers: {
  type: [
    {
      type: Schema.Types.ObjectId,
      ref: "Camper",
    }
  ],
  default: []
}
```